### PR TITLE
[lua] Correct edge case despawn logic for Haty/Vran

### PR DESCRIPTION
**Pull request by Rosienaikanaro**
Haty/Vran can be up on server launch outside of the correct moon phase. They should forcibly despawn if the phase is incorrect.

**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Haty / Bendigeit Vran can pop on the first night after server spinup but will not depop. This creates an edge case where both can be up on a night that does not meet the requirements to spawn either on server launch. 

## Steps to test these changes

!gotoid 17220000 and/or !gotoid 17220001. 
They should no longer both be valid on the first night after server spin up. At most the relevant one to the current moon phase will remain spawned.

### CODE REVIEW

File: scripts/zones/Konschtat_Highlands/mobs/Bendigeit_Vran.lua [hunk 1]
```diff
@@ -13,7 +13,11 @@ end
 
 entity.onMobRoam = function(mob)
     local hour = VanadielHour()
-    if hour >= 5 and hour < 17 then
+    local phase = VanadielMoonPhase()
+    if
+        (hour >= 5 and hour < 17) or
+        phase > 10
```
Comment (sruon) [inline scripts/zones/Konschtat_Highlands/mobs/Bendigeit_Vran.lua:19]: Can we get a new enum started for moon phases?

Comment (Rosienaikanaro) [reply scripts/zones/Konschtat_Highlands/mobs/Bendigeit_Vran.lua:19]: I'm willing to put it together, but what would your suggestion be for creating ENUMs for the phases if the phases themselves have an upper and lower bound rather than a static value?

In situations like Lunar Cry or Ecliptic Howl it doesn't matter too much with the difference between Waxing and Waning but in other places such as Chocobo Digging it would.

Comment (Rosienaikanaro) [reply scripts/zones/Konschtat_Highlands/mobs/Bendigeit_Vran.lua:19]: Per our brief discussion on discord, see issue https://github.com/LandSandBoat/server/issues/8581 for a description of the issue and the proper path forward for handling this. 

I will attempt to take on fixing the moon phase thing a little later down the line if no one else does. It shouldn't be necessary for this PR to conform with the current method.

**Review by Xaver-DaRed — APPROVED** [review]

File: scripts/zones/Konschtat_Highlands/mobs/Bendigeit_Vran.lua (hunk 1 above)
Comment (Xaver-DaRed) [reply scripts/zones/Konschtat_Highlands/mobs/Bendigeit_Vran.lua:19]: I consider this out of scope for this PR
